pg_lakehouse/src/duckdb/csv.rs

```diff
@@ -0,0 +1,515 @@
+use std::collections::HashMap;
+
+use super::utils;
+
+pub enum CsvOption {
+    AllVarchar,
+    AllowQuotedNulls,
+    AutoDetect,
+    AutoTypeCandidates,
+    Columns,
+    Compression,
+    Dateformat,
+    DecimalSeparator,
+    Delim,
+    Escape,
+    Filename,
+    Files,
+    ForceNotNull,
+    Header,
+    HivePartitioning,
+    HiveTypes,
+    HiveTypesAutocast,
+    IgnoreErrors,
+    MaxLineSize,
+    Names,
+    NewLine,
+    NormalizeNames,
+    NullPadding,
+    Nullstr,
+    Parallel,
+    Quote,
+    SampleSize,
+    Sep,
+    Skip,
+    Timestampformat,
+    Types,
+    UnionByName,
+}
+
+impl CsvOption {
+    pub fn as_str(&self) -> &str {
+        match self {
+            Self::AllVarchar => "all_varchar",
+            Self::AllowQuotedNulls => "allow_quoted_nulls",
+            Self::AutoDetect => "auto_detect",
+            Self::AutoTypeCandidates => "auto_type_candidates",
+            Self::Columns => "columns",
+            Self::Compression => "compression",
+            Self::Dateformat => "dateformat",
+            Self::DecimalSeparator => "decimal_separator",
+            Self::Delim => "delim",
+            Self::Escape => "escape",
+            Self::Filename => "filename",
+            Self::Files => "files",
+            Self::ForceNotNull => "force_not_null",
+            Self::Header => "header",
+            Self::HivePartitioning => "hive_partitioning",
+            Self::HiveTypes => "hive_types",
+            Self::HiveTypesAutocast => "hive_types_autocast",
+            Self::IgnoreErrors => "ignore_errors",
+            Self::MaxLineSize => "max_line_size",
+            Self::Names => "names",
+            Self::NewLine => "new_line",
+            Self::NormalizeNames => "normalize_names",
+            Self::NullPadding => "null_padding",
+            Self::Nullstr => "nullstr",
+            Self::Parallel => "parallel",
+            Self::Quote => "quote",
+            Self::SampleSize => "sample_size",
+            Self::Sep => "sep",
+            Self::Skip => "skip",
+            Self::Timestampformat => "timestampformat",
+            Self::Types => "types",
+            Self::UnionByName => "union_by_name",
+        }
+    }
+
+    pub fn is_required(&self) -> bool {
+        match self {
+            Self::AllVarchar => false,
+            Self::AllowQuotedNulls => false,
+            Self::AutoDetect => false,
+            Self::AutoTypeCandidates => false,
+            Self::Columns => false,
+            Self::Compression => false,
+            Self::Dateformat => false,
+            Self::DecimalSeparator => false,
+            Self::Delim => false,
+            Self::Escape => false,
+            Self::Filename => false,
+            Self::Files => true,
+            Self::ForceNotNull => false,
+            Self::Header => false,
+            Self::HivePartitioning => false,
+            Self::HiveTypes => false,
+            Self::HiveTypesAutocast => false,
```

Comment:
Do we support Hive? I've had a few users request it

Reply:
Yes